doc/source/whatsnew/v0.25.0.rst

```diff
@@ -43,6 +43,7 @@ Other API Changes
 Deprecations
 ~~~~~~~~~~~~
 
+- :func:`Series.str.replace`, when pat is single special regex character and regex is not defined, regex is by default ``False`` for now, but this might be deprecated in the future. (:issue:`24804`)
```

Comment:
use double back-ticks around ``pat``.

..is single special regex character'.. (such as ``.|\`` etc).